doc/content/integrations/pubsub/message-types/_index.md

```diff
@@ -0,0 +1,28 @@
+---
+title: "Message Types"
+description: ""
+---
+
+The Application Server can be configured to publish or subscribe to messages for any of the following events:
+- Uplink Message
+- Join Accept
+- Downlink Ack
+- Downlink Nack
+- Downlink Sent
+- Downlink Failed
+- Downlink Queued
+- Location Solved
+- Downlink Push
+- Downlink Replace
```

Comment:
Please split our publish vs subscribe

Also add "Service Data" (coming in #2541)

Reply:
Please sort

Reply:
Also push and replace are queue operations, so "Downlink Queue Push" and "... Replace"